code/__HELPERS/unsorted.dm

```diff
@@ -721,11 +721,12 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/pixel_y_offset = AM.pixel_y + M.get_y_shift()
 
 	//Irregular objects
-	if(AM.bound_height != world.icon_size || AM.bound_width != world.icon_size)
-		var/icon/AMicon = icon(AM.icon, AM.icon_state)
+	var/icon/AMicon = icon(AM.icon, AM.icon_state)
+	var/icon/AMiconheight = AMicon.Height()
```

Comment:
i'll do a test, but i think i might be able to use `_icon_height` on the actual `.icon` `RSCFILE` object and avoid the need to convert to `/icon/` in general, and that alone should shave a bit of overhead